app/main.py

```diff
@@ -1,2 +1,115 @@
+import math
+from dataclasses import dataclass
+from typing import Hashable, Any, Iterator
+
+
+@dataclass
+class Node:
+    key: Hashable
+    obj_hash: hash
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: list[Node | None] = [None] * self.capacity
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while (
+            self.hash_table[index]
+            and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+
+        return index
+
+    @property
+    def current_max_size(self) -> int:
+        return math.floor(self.capacity * 2 / 3)
+
+    def _resize(self) -> None:
+        old_hash_table = self.hash_table
+
+        self.__init__(self.capacity * 2)
+
+        for node in old_hash_table:
+            if node:
+                self.__setitem__(node.key, node.value)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            if self.size + 1 >= self.current_max_size:
+                self._resize()
+                return self.__setitem__(key, value)
+            self.size += 1
+
+        self.hash_table[index] = Node(key, hash(key), value)
+
+    def __getitem__(self, item: Any) -> Any:
+        index = self._calculate_index(item)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find object with key -> {item}")
+
+        return self.hash_table[index].value
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find value for key: {key}")
+
+        self.hash_table[index] = None
+        self.size -= 1
```

Comment:
you need to recalculate remaining positions after deleting an element because indexes may be wrong for elements that had collisions

Reply:
This will not pass the optional test on delete item - after deletion you need to recount indexes and reset all remaining elements in hash table to be sure data is correct